presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -393,7 +397,7 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
         if (!isStatisticsEnabled(session)) {
             return EMPTY_STATISTICS;
         }
-        List<HivePartition> hivePartitions = partitionManager.getPartitions(metastore, tableHandle, constraint).getPartitions();
+        List<HivePartition> hivePartitions = partitionManager.getPartitions(metastore, tableHandle, constraint).getPartitionsAsList(maxPartitions);
```

Comment:
> move `getPartitionsAsList` [...] method to this class?

+1

Reply:
> We could. Would that be nicer? IDK.

Yes. It will solve also this comment: https://github.com/prestodb/presto/pull/10259/commits/675377d3ae1db55773db42cdedce426d50cd0be1#r177652107